### ISSUE TITLE
Crash with initModel

### ISSUE THREAD

**Issue by pommedeterresautee** [facebookresearch/StarSpace#94]
When initializing with fastText pretrained (by Facebook team) model, I always get this error:
```
Arguments: 
lr: 0.05
dim: 300
epoch: 5
maxTrainTime: 8640000
saveEveryEpoch: 0
loss: hinge
margin: 0.05
similarity: cosine
maxNegSamples: 10
negSearchLimit: 50
thread: 11
minCount: 30
minCountLabel: 1
label: __label__
ngrams: 1
bucket: 2000000
adagrad: 1
trainMode: 3
fileFormat: labelDoc
normalizeText: 0
dropoutLHS: 0
dropoutRHS: 0
Start to load a trained starspace model.
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
./learn.sh : ligne 21 :  2695 Abandon                 (core dumped) ./starspace train -trainFile ./data/q_starspace.txt -model ./data/q_learned -trainMode 3 -initRandSd 0.01 -adagrad true -ngrams 1 -lr 0.05 -margin 0.05 -epoch 5 -thread 11 -dim 300 -negSearchLimit 50 -fileFormat labelDoc -similarity "cosine" -minCount 30 -verbose true -normalizeText 0 -initModel ./data/wiki.fr.head.50k.vec
```

`wiki.fr.head.50k.vec` are the first 50K lines (I tried with full file, and with first 10K ).
I have 128Gb of ram.
Training without init the model works with pretrained file works.

What may I try to init the model and avoid this error?

**Comment by pommedeterresautee** [comment]
I just noticed that fastText/word2vec txt model is different from starSpace:

* use space between word and vector
* first line contains number of words and vector size

I think it would be really great if word2vec format was managed...

**Comment by ledw** [comment]
@pommedeterresautee Thanks! You can use the word2vec / fastText format model in StarSpace. To do that, you need to change the file name to "xxx.tsv" instead of "xxx.vec" (and yes, it takes space separated word and vector). However, please specify the correct dimension of your pre-trained model by -dim arg.

**Comment by pommedeterresautee** [comment]
Thanks a lot, it works.
I close the issue.
Reopen it if you prefer to keep a trace that there is this small bug.

**Comment by sweatyrichard** [comment]
@pommedeterresautee How did it work out for you using the French pretrained Wikipedia vectors?

**Comment by pommedeterresautee** [comment]
You need to convert to the right format first, after it works.

**Comment by sweatyrichard** [comment]
@pommedeterresautee I suppose I was asking less about this specific issue and more about the differences you noticed in your application by training on top of the pretrained vectors. Technically I understand what is happening while performing additional training, but I'm missing examples of how and in which circumstances you see the gains from this.

**Comment by pommedeterresautee** [comment]
in case of very small dataset, you prefer to init with something than random. Even if the semantic init of Word embeddings doesn t exactly match your need for the task to perform
